modules/carto/src/api/parse-map.ts

```diff
@@ -16,14 +16,14 @@ import {
 import PointLabelLayer from '../layers/point-label-layer';
 import {CollisionFilterExtension} from '@deck.gl/extensions';
 import {assert} from '../utils';
-import {MapDataset, MapLayerConfig, VisualChannels} from './types';
+import {KeplerMapConfig, MapDataset, MapLayerConfig, VisualChannels} from './types';
 
 const collisionFilterExtension = new CollisionFilterExtension();
 
 export function parseMap(json) {
   const {keplerMapConfig, datasets, token} = json;
   assert(keplerMapConfig.version === 'v1', 'Only support Kepler v1');
-  const {mapState, mapStyle} = keplerMapConfig.config;
+  const {mapState, mapStyle} = keplerMapConfig.config as KeplerMapConfig;
```

Comment:
Optional: On line 36 below, if you add...

```javascript
/** @deprecated Use 'basemap'. */
mapStyle,
```

... then the deprecation is visible to TypeScript users.

Probably we should add type definitions for the fetchMap return type at some point, but doesn't need to be in this PR.

Reply:
I've added `FetchMapResult` too and it contains brief docs too.